boxoffice/models/order.py

```diff
@@ -1,38 +1,89 @@
-from decimal import Decimal
-from boxoffice.models import db, BaseMixin
-from boxoffice.models.event import Event
-from boxoffice.models.user import User
-
-__all__ = ['Order', 'HALF_DOZEN']
-
-HALF_DOZEN = [
-    [0, "0"],
-    [1, "1"],
-    [2, "2"],
-    [3, "3"],
-    [4, "4"],
-    [5, "5"],
-    [6, "6"]
-]
+from datetime import datetime
+import decimal
+from collections import namedtuple
+from boxoffice.models import db, BaseMixin, User
+from coaster.utils import LabeledEnum, buid
+from baseframe import __
 
-class Order(BaseMixin, db.Model):
+__all__ = ['Order', 'ORDER_STATUS']
+
+
+class ORDER_STATUS(LabeledEnum):
+    PURCHASE_ORDER = (0, __("Purchase Order"))
+    SALES_ORDER = (1, __("Sales Order"))
+    INVOICE = (2, __("Invoice"))
+    CANCELLED = (3, __("Cancelled Order"))
+
+
+def get_latest_invoice_no(organization):
     """
-    An Order begins with an intent to register 
-    Belongs to an Event
-    Contains all Attendees, Has an Owner
+    Returns the last invoice number used, 0 if no order has ben invoiced yet.
     """
-    __tablename__ = 'order'
+    order = Order.query.filter_by(organization=organization,
+        status=ORDER_STATUS.SALES_ORDER).order_by('created_at desc').first()
+    return order.invoice_no if order else 0
```

Comment:
Beware of `created_at`'s precision coming back to bite. It is perfectly possible to have two orders with the exact same `created_at` timestamp. Why not query for `MAX(invoice_no)`?